src/webgpu/shader/validation/expression/call/builtin/atomics.spec.ts

```diff
@@ -0,0 +1,66 @@
+export const description = `
+Validation tests for atomic builtins.
+`;
+
+import { makeTestGroup } from '../../../../../../common/framework/test_group.js';
+import { ShaderValidationTest } from '../../../shader_validation_test.js';
+
+export const g = makeTestGroup(ShaderValidationTest);
+
+const atomicOps = [
+  'atomicAdd(&a, 1)',
+  'atomicSub(&a, 1)',
+  'atomicMax(&a, 1)',
+  'atomicMin(&a, 1)',
+  'atomicAnd(&a, 1)',
+  'atomicOr(&a, 1)',
+  'atomicXor(&a, 1)',
+  'atomicLoad(&a)',
+  'atomicStore(&a, 1)',
+];
+
+g.test('stage')
+  .specURL('https://www.w3.org/TR/WGSL/#atomic-rmw')
+  .desc(
+    `
+Atomic built-in functions must not be used in a vertex shader stage.
+`
+  )
+  .params(u =>
+    u
+      .combine('stage', ['fragment', 'vertex', 'compute'] as const) //
```

Comment:
nit, I don't think we typically do the `//` to force breaking in the CTS and just let the formatter format is it desires.

Reply:
I much prefer it, as it's way easier to see what each parameter is.